Cargador.cpp

```diff
@@ -0,0 +1,53 @@
+Cargador::Cargador()
+{
+
+}
+
+void Cargador::cargadorDeABB(ABB<Diccionario> &arbolAeropuertos , ifstream archivo)
+{
+    //char[3] o string para el codigo IATA?
+    string nombre ,ciudad ,pais,codigoIATA ;
+    double superficie;
+    int cantidadTerminales,destinosNacionales,destinosInternacionales ;
+
+
+    if(!archivo.fail())
+    {
+        while(!archivo.eof())
+        {
+            //Puntero a los datos del aeropuerto
+            Aeropuerto *pAeropuerto ;
+
+            archivo >> codigoIATA;
+            archivo >> nombre;
+            archivo >> ciudad;
+            archivo >> pais;
+            archivo >> superficie;
+            archivo >> cantidadTerminales;
+            archivo >> destinosNacionales;
+            archivo >> destinosInternacionales;
+
+            pAeropuerto = new Aeropuerto(codigoIATA,nombre,ciudad,pais,superficie,cantidadTerminales,destinosNacionales,destinosInternacionales);
+
+            //Creo un diccionario y le paso por parametro el puntero y la clave del codigo
+            Diccionario dicAeropuerto(codigoIATA,pAeropuerto);
+
+            //Inserto el nodo en el arbol
+            arbolAeropuertos.insertar(dicAeropuerto);
+        }
+        cout << "Se han cargado los Aeropuertos en el arbol correctamente " << endl;
+
+    }
+    else{
+        //Tiro la excepcion
+        try {
+
+            throw(Excepcion(ExcepcionABB));
+        }
+        catch(Excepcion& e) {
+            cout << e.what() << endl;
+        }
```

Comment:
- Idem comentarios anteriores, esto lo haría afuera.
- No la tengo muy clara con excepciones, tengo que repasar eso. Pero no entiendo para qué esta el bloque `try`, no se puede sacar? Y dejar algo como:
```suggestion
        throw(Excepcion(ExcepcionABB));
        catch(Excepcion& e) {
            cout << e.what() << endl;
        }
```
Mi razonamiento es el siguiente: si no se pudo abrir el archivo, para qué quiero _intentar lanzar una excepción_? No tendría que lanzarla directamente? O sino en vez de el bloque `if/else` **y** el bloque `try/catch`, no se podría poner solo el `try/catch`?